src/App.tsx

```diff
@@ -1,24 +1,221 @@
-
-const USER_ID = 0;
+import { filteredTodos } from './utils/filter';
+import { INITIAL_STATE_TEMPTODO } from './constants/initial_state_newTodo';
+import { USER_ID } from './constants/user_id';
+import { ErrorMessage } from './components/ErrorMessage';
+import { Error } from './enums/Error';
+import { handlerError } from './utils/Errors';
+import { client } from './utils/fetchClient';
+import { TodoForm } from './components/TodoForm';
+import { reducer } from './reducer';
+import { ReducerType } from './enums/Reducer';
 
 export const App: React.FC = () => {
+  const [filter, setFilter] = useState(Filter.ALL);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [tempTodo, dispatch] = useReducer(reducer, INITIAL_STATE_TEMPTODO);
+  const [processingIds, setProcessingIds] = useState<number[]>([]);
+  const [isCreated, setIsCreated] = useState(false);
+  const [isError, setIsError] = useState<Error>(Error.RESET);
+  const [isArrow, setIsArrow] = useState(false);
+
+  const isProcessing = (id: number) => {
+    if (processingIds.includes(id) || id === 0) {
+      return true;
+    }
+
+    return false;
+  };
+
+  const ErrorTitle = handlerError(isError);
+
+  const onShowArrow = useMemo(() => {
+    return todos.every(todo => todo.completed)
+      ? setIsArrow(true)
+      : setIsArrow(false);
+  }, [todos]);
+
+  const visibleTodos = useMemo(() => {
+    return filteredTodos(todos, filter);
+  }, [todos, filter]);
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then(setTodos)
+      .catch(() => {
+        setIsError(Error.UPLOAD);
+      });
+  }, []);
+
+  const onAdd = (todo: Todo) => {
+    const {
+      title, userId, completed,
+    } = todo;
+
+    setIsCreated(true);
+
+    return client.post<Todo>('/todos', { title, userId, completed })
+      .then(result => {
+        setTodos(prev => [...prev, result]);
+        dispatch({ type: ReducerType.RESET });
+      })
+      .catch(() => setIsError(Error.ADD))
+      .finally(() => setIsCreated(false));
+  };
+
+  const onDelete = (todoIdToDelete: number) => {
+    setProcessingIds(prev => [...prev, todoIdToDelete]);
+
+    removeTodo(todoIdToDelete)
+      .then(() => getTodos(USER_ID))
+      .then(setTodos)
+      .catch(() => setIsError(Error.REMOVE))
+      .then(() => setProcessingIds((prev) => {
+        return prev.filter(id => id !== todoIdToDelete);
+      }));
+  };
+
+  const onUpdate = (todoID: number, data: Partial<Todo>) => {
+    setProcessingIds(prev => [...prev, todoID]);
+
+    return updateTodo(todoID, data)
+      .then(() => setProcessingIds((prev) => {
+        return prev.filter(id => id !== todoID);
+      }))
+      .catch(() => setIsError(Error.UPDATE))
+      .then(() => getTodos(USER_ID))
+      .then(setTodos);
+  };
+
+  const handleSubmit = (e: React.FormEvent) => {
+    e.preventDefault();
+
+    if (!tempTodo.title.trim()) {
+      return setIsError(Error.TITLE);
+    }
+
+    return onAdd(tempTodo);
+  };
+
+  const onClear = () => {
```

Comment:
if you put some function to the other component - use useCallback with the function.
Bc your function creates every render, accordingly, your child component renders to
Use it in all cases
```suggestion
  const onClear = useCallback() => {
```